cmd/fetchd/cmd/genasiupgrade.go

```diff
@@ -41,23 +35,54 @@ const (
 	ValAddressPrefix  = "valoper"
 	ConsAddressPrefix = "valcons"
 
-	NewBaseDenom   = "asi"
-	NewDenom       = "aasi"
-	NewAddrPrefix  = "asi"
-	NewChainId     = "asi-1"
-	NewDescription = "ASI Token" // TODO(JS): change this, potentially
-
-	OldDenom      = "afet"
+	NewAddrPrefix = "asi"
 	OldAddrPrefix = "fetch"
 )
 
+var networkInfos = map[string]NetworkConfig{
+	"mainnet": {
```

Comment:
As you suggested during the discussion, the key (the `"mainnet" in this particular example)` should have value of original `chain-id` from input genesis.json (e.g. `fetchhub-4` in this particular case).